### ISSUE TITLE
Deferred Loading still doesn't work properly in SSR mode

### ISSUE THREAD

**Issue by artem-zur** [SAP/spartacus#8089]
## Overview

Deferred Loading makes double loading (screen blinking) during first page rendering in SSR mode.

## Steps to Reproduce

1. Activate deferredLoading inside layout configuration with the following parameters: 
```
deferredLoading: {
  strategy: DeferLoadingStrategy.DEFER,
  intersectionMargin: '50px'
}
```
2. Build and start application in SSR mode.
3. Open any page in Browser with cleared cache.
4. Page loads twice. Looks like SPA application reloads view rendered by SSR.

## Expected behaviour

Page correctly loads without screen blinking.

## Environment Details

- Spartacus:  *-2.0.0
- Desktop browser: Chrome
- Other environment details: Node.js 10.16.0

## Additional context

P.S. See original issue #6449 for previous Spartacus version.

**Comment by tobi-or-not-tobi** [comment]
@artem-zur just checking, can this be related to the page layout? We have mobile vs desktop page layouts in standard Spartacus for the header of the page. Since Spartacus/SSR doesn't have device detection as of now, it means that SSR will always produce the mobile version. If you load this on a larger device (desktop), the layout will be rebuild.

**Comment by artem-zur** [comment]
@tobi-or-not-tobi hi!

I'm so sorry for such long delay with the answer. Honestly saying I'm not sure about page layout cuz I can reproduce the issue in mobile version also.

**Comment by dunqan** [comment]
It was already fixed for 2.1, we had a regression in 3.0 which was fixed by #10718.
